src/vs/platform/tunnel/common/tunnel.ts

```diff
@@ -146,13 +146,26 @@ export function extractLocalHostUriMetaDataForPortMapping(uri: URI): { address:
 		return undefined;
 	}
 	const localhostMatch = /^(localhost|127\.0\.0\.1|0\.0\.0\.0):(\d+)$/.exec(uri.authority);
```

Comment:
Another way to consider, instead of checking each query string parameter separately, maybe you can check them all together (assuming that they are indeed unescaped):

```typescript
const toSearch = uri.authority + uri.query;
const localhostMatch = /^(localhost|127\.0\.0\.1|0\.0\.0\.0):(\d+)$/.exec(toSearch);
```